runtime/compiler/z/codegen/S390J9CallSnippet.cpp

```diff
@@ -1460,7 +1460,18 @@ TR::S390JNICallDataSnippet::emitSnippetBody()
       // _targetAddress/function pointer of native method
       *(intptr_t *) cursor = (intptr_t) _targetAddress;
       TR_OpaqueMethodBlock *method = getNode()->getSymbol()->castToResolvedMethodSymbol()->getResolvedMethod()->getPersistentIdentifier();
-      TR_PatchJNICallSite::make(cg()->fe(), cg()->trPersistentMemory(), (uintptr_t) method, cursor, comp->getMetadataAssumptionList());
+
+      if (comp->isOutOfProcessCompilation())
+         {
+         // For JITServer we need to build a list of assumptions that will be sent to client at end of compilation
+         intptr_t offset = cursor - cg()->getCodeStart();
```

Comment:
Is `cg()->getCodeStart()` the beginning of the code cache (thus including all of the prologue, pre-prologue and code cache header?
On x86 we use `self()->getBinaryBufferStart()`

Reply:
A similar thing is done inside `J9::CodeGenerator::registerAssumptions()` in this PR: https://github.com/eclipse/openj9/pull/7948/files. 

But it looks like the call has since been replaced with `self()->getBinaryBufferStart()`. On Z we should be able to make the same call.

Reply:
The explanation for why it was changed is provided here, and I think we should do the same on Z:
https://github.com/eclipse/openj9/commit/b61f2c8f2ce36ba3db652d4a557a3cac24de2e2a

Reply:
Fixed here: dbfa57d